File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
//  Root, Left, Right
class Solution {
    private List<Integer> result;

    public List<Integer> preorderTraversal(TreeNode root) {
        result = new ArrayList<>();
        preorder(root);
        return result;
    }

    private void preorder(TreeNode node) {
        if (node == null) {
            return;
        }
        result.add(node.val);
        // First value we add, because the pre-order traversal dictates that:
        // Root, Left, Right 
        preorder(node.left);
        preorder(node.right);
    }
}

File: Data Structures & Algorithms/special-array-i/submission-0.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        
        i = 0
        while i != len(nums) - 1:
            first = nums[i]
            sec = nums[i+1]
            if (first % 2 == 0 and sec % 2 == 0) or (first % 2 == 1 and sec % 2 == 1):
                return False
            i += 1
        return True
            
        